fuel-vm/src/error.rs

```diff
@@ -5,45 +5,50 @@ use fuel_asm::{
     PanicReason,
     RawInstruction,
 };
+use fuel_storage::StorageError;
 use fuel_tx::CheckError;
+
+#[cfg(feature = "std")]
 use thiserror::Error;
```

Comment:
Maybe we want to use https://docs.rs/thiserror-no-std/latest/thiserror_no_std/

Reply:
That crate looks incredibly suspicious. It contains no explanation on differences with `thiserror` crate itself, it's owned by a different user than `thiserror`, and all links (docs, repo, ...) point to those of the original `thiserror`.

We don't really need `thiserror` here at all, but I decided to keep it to avoid unnecessary breaking changes.

Reply:
Removing `thiserror` and implementing `Display` would be a nice simplification. 

I am okay with the current if we would like to remove `thiserror` in a separate PR.